docs/transactions/source/transaction-validation.rst

```diff
@@ -1,34 +1,113 @@
 Transaction Validation
 ======================
 
-If a transaction satisfies the constraints listed below,
+If a transaction satisfies the constraints (or rules) listed below,
 then it is considered valid.
-The process of checking those constraintst is called transaction validation.
+The process of checking those constraints is called transaction validation.
+
+Each version X.Y.Z of the IPDB Transaction Spec may have different constraints.
+That is, the constraints may change from one version to the next.
+
+
+How Validation Code Decides Which Version to Use
+------------------------------------------------
+
+When given a transaction to validate,
+validation code should check the value of ``"version"``
+inside the transaction.
+The valid values are those with a corresponding set of JSON Schema files
+(which can be found in the ``tx_schema`` directory
+of the `IPDB Transaction Spec repository
+<https://github.com/ipdb/ipdb-tx-spec>`_).
+If ``"version"`` doesn't have a valid value, then the transaction is invalid.
+Otherwise, the transaction should be validated according
+to the validation constraints described in that version of the IPDB Transaction Spec.
 
 
 JSON Schema Validation
 ----------------------
 
-When checking the validity of a transaction,
-an IPDB-compliant server
-must check it against a formal schema
+JSON Schema Validation is done by checking the transaction against
+a formal `JSON Schema  <http://json-schema.org/>`_
 defined in a set of `JSON Schema <http://json-schema.org/>`_ files.
 Those files can be found
 in the ``tx_schema/`` directory
-of the IPDB Transaction Spec repository.
+of the `IPDB Transaction Spec repository
+<https://github.com/ipdb/ipdb-tx-spec>`_.
 
-Tip 1: There's a nice explanation of JSON Schema in the website
-`"Understanding JSON Schema"
-<https://spacetelescope.github.io/understanding-json-schema/index.html>`_.
+.. note::
 
-Tip 2: Python 3 code for checking a transaction against those JSON Schema files
-can be found in the `source code of BigchainDB Server
-<https://github.com/bigchaindb/bigchaindb>`_.
-At the time of writing, it was in the file
-``bigchaindb/common/schema/__init__.py``.
+   Tip 1: There's a nice explanation of JSON Schema in the website
+   `"Understanding JSON Schema"
+   <https://spacetelescope.github.io/understanding-json-schema/index.html>`_.
+
+   Tip 2: Python 3 code for checking a transaction against JSON Schema files
+   can be found in the `source code of BigchainDB Server
+   <https://github.com/bigchaindb/bigchaindb>`_.
+   At the time of writing, it was in the file
+   ``bigchaindb/common/schema/__init__.py``.
 
 
 Other Constraints
 -----------------
 
-TODO
+The output.amount Rule
+^^^^^^^^^^^^^^^^^^^^^^
+
+For all outputs, once output.amount has been converted
+from a string to an integer,
+it must be between 1 and 9×10¹⁸, inclusive.
+The reason for the upper bound is to keep amount within what a server
+can comfortably represent using a 64-bit signed integer,
```

Comment:
Now that I'm thinking, why we are using a `signed integer`? An amount cannot be negative. If we use an `unsigned integer` we would be able to double maximum amount. This comment is out of the scope of this PR, maybe I should make a formal proposal to increase the upper limit to `2^64`, what do you think?

Reply:
It's up to implementors to decide what kind of data type to use for the amount internally, after converting it from a string to a number. (It's a string when being sent as JSON over the wire or when being stored in the backend database.) I guess Scott's thinking was that 2^63 is "playing it safe."